src/lib/components/ScheduleWidget/ScheduleWidgetList.svelte

```diff
@@ -2,31 +2,90 @@
 	import type { EventSchedule, Schedule } from '$lib/types/Schedule';
 	import { useI18n } from '$lib/i18n';
 	import { upperFirst } from '$lib/utils/upperFirst';
-	import { format, isSameDay } from 'date-fns';
+	import { addDays, format, isSameDay } from 'date-fns';
 
 	export let schedule: Schedule;
 	let className = '';
 	export { className as class };
 
 	const { t, locale } = useI18n();
+
+	$: dateFilter = '';
+	$: locationFilter = '';
+	$: nameFilter = '';
+	$: descriptionFilter = '';
 	let scheduleEventByDay: Record<string, EventSchedule[]> = schedule.events.reduce(
 		(acc, event) => {
-			let dateKey = format(new Date(event.beginsAt), 'yyyy-MM-dd');
+			const startDate = new Date(event.beginsAt);
+			const endDate = event.endsAt ? new Date(event.endsAt) : startDate;
+
+			if (isNaN(startDate.getTime())) {
+				return acc;
+			}
+			if (isNaN(endDate.getTime())) {
+				return acc;
+			}
+
+			let currentDate = new Date(startDate);
+			while (currentDate <= endDate) {
+				const dateKey = format(currentDate, 'yyyy-MM-dd');
+
+				if (!acc[dateKey]) {
+					acc[dateKey] = [];
+				}
+
+				acc[dateKey].push(event);
+
+				currentDate = addDays(currentDate, 1);
+			}
 
-			if (!acc[dateKey]) {
-				acc[dateKey] = [];
+			return acc;
+		},
+		{} as Record<string, EventSchedule[]>
+	);
```

Comment:
Ok I let this go before, but you shouldn't use `.reduce()`, use `fromEntries(...)` instead it's more readble/understandable